src/librustc_resolve/macros.rs

```diff
@@ -166,20 +166,16 @@ impl<'a> base::Resolver for Resolver<'a> {
         self.whitelisted_legacy_custom_derives.contains(&name)
     }
 
-    fn visit_expansion(&mut self, mark: Mark, expansion: &Expansion, derives: &[(Mark, Path)]) {
+    fn visit_expansion(&mut self, mark: Mark, expansion: &Expansion, derives: &[Mark]) {
         let invocation = self.invocations[&mark];
         self.collect_def_ids(mark, invocation, expansion);
 
         self.current_module = invocation.module.get();
-        {
-            let mut unresolved = self.current_module.unresolved_invocations.borrow_mut();
-            unresolved.remove(&mark);
-            for &(derive, _) in derives {
-                unresolved.insert(derive);
-                self.invocations.insert(derive, invocation);
-            }
```

Comment:
Hmm, this commit also reverts some changes done in the original PR.

Reply:
This and the other nonsensical changes are probably due to a naive rebase where I just applied all nonconflicting changes from both sides. I'll have to go through and fix these or redo the rebase.